File: server/webflix/src/main/java/com/instil/webflix/voucher/service/VoucherService.java
package com.instil.webflix.voucher.service;


import java.util.Date;

import java.util.List;

import com.instil.webflix.voucher.model.Voucher;

public interface VoucherService {
	boolean getVoucherValid(String name);

	void createVoucher(String code, String discount, Date expiryDate);
	void checkForExpiredGlobal();

	
	List<Voucher> getAllVouchers();

}

File: client/webflix/src/app/service/voucher/voucher.service.ts
import {Observable} from 'rxjs/Observable';
import {Voucher} from '../../model/voucher';

export abstract class VoucherService {
  abstract getVoucherValid(name: String): Observable<Voucher>;

  abstract getAllVouchers(): Observable<Voucher[]>;


  abstract removeVoucher(voucher: Voucher): Observable<boolean>;
  abstract toggleGlobalVoucher(voucher: Voucher, global: boolean): Observable<boolean>;
  abstract createVoucher(code: string, discount: string, expiryDate: Date):  Observable<boolean>;


  abstract getAllGlobalVouchers(): Observable<Voucher[]>;

  abstract getUsedVouchers(): Observable<number[]>;

  abstract addUsedVoucher(voucherId: number): Observable<boolean>;
}

